src/Umbraco.Core/PropertyEditors/DataValueReferenceFactoryCollection.cs

```diff
@@ -4,64 +4,148 @@
 
 namespace Umbraco.Cms.Core.PropertyEditors;
 
+/// <summary>
+/// Provides a builder collection for <see cref="IDataValueReferenceFactory" /> items.
+/// </summary>
 public class DataValueReferenceFactoryCollection : BuilderCollectionBase<IDataValueReferenceFactory>
 {
+    // TODO: We could further reduce circular dependencies with PropertyEditorCollection by not having IDataValueReference implemented
+    // by property editors and instead just use the already built in IDataValueReferenceFactory and/or refactor that into a more normal collection
```

Comment:
A quick note on this comment (that I only moved to the top of the class): refactoring all property editors to separate `IDataValueReferenceFactory` implementations would separate out the responsibility of getting the references, reduce the dependency tree and avoid having to create value editors to check whether it implements the required interface. It would also allow removing the default implementation or replace it with a custom one if you'd like...

We would first need to rewrite our own `IDataValueReferenceFactory` implementations for all built-in/supported property editors and then remove the `dataEditor.GetValueEditor() is IDataValueReference dataValueReference` lookup/check within this class. This would be a breaking change, as custom implementations would need to do the same. Rewriting without removing this lookup wouldn't give any benefits, because the value editors would still be created to do this check 🤓